frame/uniques/src/benchmarking.rs

```diff
@@ -0,0 +1,429 @@
+		let target_lookup = T::Lookup::unlookup(target);
+		assert!(Assets::<T, I>::mint(origin.clone().into(), Default::default(), target_lookup, 100u32.into()).is_ok());
+	}
+	swap_is_sufficient::<T, I>(&mut s);
+}
+
+fn add_sufficients<T: Config<I>, I: 'static>(minter: T::AccountId, n: u32) {
+	let origin = SystemOrigin::Signed(minter);
+	let mut s = true;
+	swap_is_sufficient::<T, I>(&mut s);
+	for i in 0..n {
+		let target = account("sufficient", i, SEED);
+		let target_lookup = T::Lookup::unlookup(target);
+		assert!(Assets::<T, I>::mint(origin.clone().into(), Default::default(), target_lookup, 100u32.into()).is_ok());
+	}
+	swap_is_sufficient::<T, I>(&mut s);
+}
+
+fn add_approvals<T: Config<I>, I: 'static>(minter: T::AccountId, n: u32) {
+	T::Currency::deposit_creating(&minter, T::ApprovalDeposit::get() * n.into());
+	let minter_lookup = T::Lookup::unlookup(minter.clone());
+	let origin = SystemOrigin::Signed(minter);
+	Assets::<T, I>::mint(
+		origin.clone().into(),
+		Default::default(),
+		minter_lookup,
+		(100 * (n + 1)).into(),
+	).unwrap();
+	for i in 0..n {
+		let target = account("approval", i, SEED);
+		T::Currency::make_free_balance_be(&target, T::Currency::minimum_balance());
+		let target_lookup = T::Lookup::unlookup(target);
+		Assets::<T, I>::approve_transfer(
+			origin.clone().into(),
+			Default::default(),
+			target_lookup,
+			100u32.into(),
+		).unwrap();
+	}
+}
+
+fn assert_last_event<T: Config<I>, I: 'static>(generic_event: <T as Config<I>>::Event) {
+	let events = frame_system::Pallet::<T>::events();
+	let system_event: <T as frame_system::Config>::Event = generic_event.into();
+	// compare to the last event record
+	let frame_system::EventRecord { event, .. } = &events[events.len() - 1];
+	assert_eq!(event, &system_event);
+}
+
+fn assert_event<T: Config<I>, I: 'static>(generic_event: <T as Config<I>>::Event) {
+	let system_event: <T as frame_system::Config>::Event = generic_event.into();
+	let events = frame_system::Pallet::<T>::events();
+	assert!(events.iter().any(|event_record| {
+		matches!(&event_record, frame_system::EventRecord { event, .. } if &system_event == event)
+	}));
+}
+
+benchmarks_instance_pallet! {
+	create {
+		let caller: T::AccountId = whitelisted_caller();
+		let caller_lookup = T::Lookup::unlookup(caller.clone());
+		T::Currency::make_free_balance_be(&caller, DepositBalanceOf::<T, I>::max_value());
+	}: _(SystemOrigin::Signed(caller.clone()), Default::default(), caller_lookup, 1u32.into())
+	verify {
+		assert_last_event::<T, I>(Event::Created(Default::default(), caller.clone(), caller).into());
+	}
+
+	force_create {
+		let caller: T::AccountId = whitelisted_caller();
+		let caller_lookup = T::Lookup::unlookup(caller.clone());
+	}: _(SystemOrigin::Root, Default::default(), caller_lookup, true, 1u32.into())
+	verify {
+		assert_last_event::<T, I>(Event::ForceCreated(Default::default(), caller).into());
+	}
+
+	destroy {
+		let c in 0 .. 5_000;
+		let s in 0 .. 5_000;
+		let a in 0 .. 5_00;
```

Comment:
generally its nice to make such constants configurable if they will affect the weight of this pallet, as different chains may have different expected limits of these constants